### PR TITLE
test(button): refactor unit tests

### PR DESCRIPTION
**Pull request by alfredskeden**
**Describe pull-request**  
This pull request refactors the unit tests for button. follow this more
https://kentcdodds.com/blog/testing-implementation-details
with less implementation details for better testing and developer experience

Also updates all images to be page specific and not component specific

**Solving issue**  
Fixes: [CDEP-2961](https://tegel.atlassian.net/browse/CDEP-2961)

**How to test**  
1. Navigate to packages/core.
2. Check the code for tests in button component.
3.  Make sure docker is running.
4. Then run `npm run test` in core to run the tests. Make sure they run.

**Additional context** 
We are still in exploration for good testing.

[CDEP-2961]: https://tegel.atlassian.net/browse/CDEP-2961

### CODE REVIEW

**Review by theJohnnyMe — APPROVED** [review]
Looks good to me buddy, nice work!

File: packages/core/src/components/button/test/basic/index.html [hunk 1]
```diff
@@ -12,6 +12,6 @@
   </head>
 
   <body>
-    <tds-button id="basicButton" text="Button"></tds-button>
+    <tds-button id="basicButton" text="Button" data-testid="tds-button-testid"></tds-button>
```
Comment (theJohnnyMe) [inline packages/core/src/components/button/test/basic/index.html:15]: Only thing you can get rid off is "id" in all index.html files accross button test. I think it is now used any more so it is not needed...

File: packages/core/src/components/button/test/basic/button.e2e.ts [hunk 2]
```diff
@@ -3,97 +3,40 @@ import { expect } from '@playwright/test';
 
 const componentTestPath = 'src/components/button/test/basic/index.html';
 
-test.describe('tds-button', () => {
+test.describe('tds-button-basic', () => {
   test('renders basic button correctly', async ({ page }) => {
     await page.goto(componentTestPath);
-    const button = page.locator('tds-button');
-    await expect(button).toHaveScreenshot({ maxDiffPixels: 0 });
+    const button = page.getByTestId('tds-button-testid');
```
Comment (timrombergjakobsson) [inline packages/core/src/components/button/test/basic/button.e2e.ts:9]: Question: should we use `getByRole` here also? As we can. Or does using `getByTestId` here provide anything else?

File: packages/core/src/components/button/test/basic/button.e2e.ts [hunk 3]
```diff
@@ -3,97 +3,40 @@ import { expect } from '@playwright/test';
...
+    await expect(button).toBeVisible();
   });
 
   test('Check so that height is correct to lg/default measurements', async ({ page }) => {
     await page.goto(componentTestPath);
-    const button = page.locator('tds-button');
-    const innerButton = button.locator('button');
-    const buttonHeight = await innerButton.evaluate((button2) => getComputedStyle(button2).height);
+    const button = page.getByText('Button', { exact: true });
+    const buttonHeight = await button.evaluate((style) => getComputedStyle(style).height);
```
Comment (timrombergjakobsson) [inline packages/core/src/components/button/test/basic/button.e2e.ts:39]: Nice!

Comment (alfredskeden) [reply packages/core/src/components/button/test/basic/button.e2e.ts:39]: Yeah not checking the height might be a breaking change in some applications using the code so needs to be checked :D

File: packages/core/src/components/button/test/basic/button.e2e.ts (hunk 2 above)
Comment (alfredskeden) [reply packages/core/src/components/button/test/basic/button.e2e.ts:9]: Erm yeah I can see few. the test id selects `tds-button` while `.getByRole('button')` would get the <button> element. So its really what we would like to test here. Check if tds-button is in the dom or if it has a <button> inside of it...

Comment (alfredskeden) [reply packages/core/src/components/button/test/basic/button.e2e.ts:9]: Updated comment didnt see the `<button>` dissapeared at first. But no problem, I guess we have to decide on a pattern here on what we would like to test on the first test with the screenshot.

Comment (timrombergjakobsson) [reply packages/core/src/components/button/test/basic/button.e2e.ts:9]: Yeah! What do you think @theJohnnyMe and @alexandereneroth?

**Pull request merged**